### PR TITLE
Strategy pattern for loading data

### PR DESCRIPTION
**Pull request by Ivikhostrup**
Skriv meget gerne om det giver mening at det ligger i sin egen folder i core.

### CODE REVIEW

File: Core/DataLoadStrategy/ComputerDataReader.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿using System.Globalization;
+using System.IO;
+using System.Linq;
+using AssetManagement.DataAccessLibrary.DataModels;
+
+namespace AssetManagement.Core.DataLoadStrategy
+{
+    public class ComputerDataReader : IAssetLoadStrategy
```
Comment (Pattrigue) [inline Core/DataLoadStrategy/ComputerDataReader.cs:8]: Please make classes sealed unless they are designed to be extensible

File: Core/DataLoadStrategy/AssetLoadContext.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
+﻿namespace AssetManagement.Core.DataLoadStrategy
+{
+    public class AssetLoadContext
```
Comment (Pattrigue) [inline Core/DataLoadStrategy/AssetLoadContext.cs:3]: Please make classes sealed unless they are designed to be extensible

File: Core/DataLoadStrategy/ComputerDataReader.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        /// <summary>
+        /// Strategy for reading computer data
+        /// </summary>
+        /// <param name="filePath">The filepath</param>
+        /// <param name="separator">The separator</param>
+        /// <returns></returns>
+        public object ReadAssetFromFile(string filePath, char separator)
```
Comment (Pattrigue) [inline Core/DataLoadStrategy/ComputerDataReader.cs:16]: Should be generic instead

File: Core/DataLoadStrategy/ComputerDataReader.cs (hunk 1 above)
Comment (laurits-mumberg) [reply Core/DataLoadStrategy/ComputerDataReader.cs]: Now sealed

File: Core/DataLoadStrategy/AssetLoadContext.cs (hunk 2 above)
Comment (laurits-mumberg) [reply Core/DataLoadStrategy/AssetLoadContext.cs]: Now sealed

File: Core/DataLoadStrategy/ComputerDataReader.cs (hunk 3 above)
Comment (laurits-mumberg) [reply Core/DataLoadStrategy/ComputerDataReader.cs:16]: Made the function ReadAssetFromFile generic. Is this what you were thinking?

File: Core/DataLoadStrategy/AssetLoadContext.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        
+        /// <summary>
+        /// Executes the method depending on which strategy was given
+        /// </summary>
+        /// <param name="filePath">The filepath</param>
+        /// <param name="separator">The separator</param>
+        /// <returns></returns>
+        public T LoadData<T>(string filePath, char separator) where T: new()
+        {
+            T result = (T)AssetLoadStrategy.ReadAssetFromFile(filePath, separator);
```
Comment (humleflue) [inline Core/DataLoadStrategy/AssetLoadContext.cs:26]: Is this always going to be a file?

File: Core/DataLoadStrategy/ComputerDataReader.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public sealed class ComputerDataReader : IAssetLoadStrategy
+    {
+        /// <summary>
+        /// Strategy for reading computer data
+        /// </summary>
+        /// <param name="filePath">The filepath</param>
+        /// <param name="separator">The separator</param>
+        /// <returns></returns>
+        public object ReadAssetFromFile(string filePath, char separator)
```
Comment (humleflue) [inline Core/DataLoadStrategy/ComputerDataReader.cs:16]: Maybe a generic stream could be given to the method to be able to handle data in a more generic way (Something Thomas was talking about)

File: Core/DataLoadStrategy/IAssetLoadStrategy.cs [hunk 6]
```diff
@@ -0,0 +1,7 @@
+﻿namespace AssetManagement.Core.DataLoadStrategy
+{
+    public interface IAssetLoadStrategy
+    {
+        object ReadAssetFromFile(string filePath, char separator);
```
Comment (humleflue) [inline Core/DataLoadStrategy/IAssetLoadStrategy.cs:5]: Maybe a generic stream could be given to the method to be able to handle data in a more generic way (Something Thomas was talking about)

**Review by humleflue — CHANGES_REQUESTED** [review]
Generally it looks fine. My biggest issue is, that right now it looks like you're committing to file data a bit too early in the process. I might be wrong though.

Nice work! :)

**Comment by laurits-mumberg** [comment]
Changed up the LoadData function, so that it no longer simply calls the given function. Previously the AssetLoadContext felt more as an extra step for calling a function, rather than an implementation of a strategy pattern. It is now specialized for csv files, which makes the use of a strategy pattern much more obvious. (IN MY HUMBLE OPINION)

**Review by Pattrigue — APPROVED** [review]
LGTM!

**Pull request merged**